placeholder_images.py

```diff
@@ -0,0 +1,65 @@
+from wmfdata.spark import get_session
+import argparse
+import requests
+import pandas as pd
+import numpy as np
+import findspark
+
+findspark.init('/usr/lib/spark2')
+
+spark = get_session(type='regular', app_name="ImageRec-DEV Training")
+
+
+class PlaceholderImages:
+    def __init__(self, snapshot):
+        self.snapshot = snapshot
+
+    @staticmethod
+    def get_placeholder_category():
+        try:
+            r = requests.get('https://petscan.wmflabs.org/', params={'psid': '18699732', 'format': 'json'})
+            r.raise_for_status()
+            response_json = r.json()
+            results = response_json['*'][0]['a']['*']
+            plhd_cat = pd.DataFrame(np.array([[r['id'], r['title']] for r in results]),
+                                    columns=['id', 'title'])
+        except requests.exceptions.RequestException:
+            return None
+        else:
+            return plhd_cat
+
+    def run(self):
+        plhd_cat = self.get_placeholder_category()
+
+        if plhd_cat is None:
+            plhd_table = 'aikochou.placeholder_category'
```

Comment:
maybe we could template the database part (`aikochou`) and make it configurable?

Reply:
I did template it to make it configurable while defaulting to `aikochou`. I didn't modify how it is being called in `publish.sh` to use another username because right now we are not storing this table but maybe we should

Reply:
No worries about `publish.sh`. Hopefully it will go away soon.